java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -2341,3 +2346,72 @@ private fun KotlinFileExtractor.extractBreakContinue(e: KtExpressionWithLabel, i
         }
     }
 }
+
+/*
+OLD KE1:
+private fun KotlinFileExtractor.getVariableLocationProvider(v: KtProperty): IrElement {
+    val init = v.initializer
+    if (v.startOffset < 0 && init != null) {
+        // IR_TEMPORARY_VARIABLEs have no proper location
+        return init
+    }
+
+    return v
+}
+*/
+
+context(KaSession)
+private fun KotlinFileExtractor.extractVariable(
+    v: KtProperty,
+    callable: Label<out DbCallable>,
+    parent: Label<out DbStmtparent>,
+    idx: Int
+) {
+    with("variable", v) {
+        val stmtId = tw.getFreshIdLabel<DbLocalvariabledeclstmt>()
+        val locId = tw.getLocation(v) // OLD KE1: getVariableLocationProvider(v))
+        tw.writeStmts_localvariabledeclstmt(stmtId, parent, idx, callable)
+        tw.writeHasLocation(stmtId, locId)
+        extractVariableExpr(v, callable, stmtId, 1, stmtId)
+    }
+}
+
+context(KaSession)
+private fun KotlinFileExtractor.extractVariableExpr(
+    v: KtProperty,
+    callable: Label<out DbCallable>,
+    parent: Label<out DbExprparent>,
+    idx: Int,
+    enclosingStmt: Label<out DbStmt>,
+    extractInitializer: Boolean = true
```

Comment:
Are we definitely going to need this in KE2? Should we leave it commented out, or at least add a TODO, if we don't know?